raiden-ts/src/transport/epics/init.ts

```diff
@@ -206,10 +189,10 @@ function fetchSortedMatrixServers$(matrixServerLookup: string, httpTimeout: numb
         response.ok,
         `Could not fetch server list from "${matrixServerLookup}" => ${response.status}`,
       );
-      return response.text();
+      return decode(MatrixServerInfo, response.json());
```

Comment:
`response.json()` returns a `Promise<any>` (to the json object), so you need to `return response.json();` (in order to get it merged to the stream and unrolled), and then, on a `map`, `decode` it.

Reply:
TS didn't complain because `decode` signature is very forgiving, since it's expected to receive anything and decode if possible or throw if it can't.

Reply:
You can also `decode + mergeMap` the array on a single step (since decode is synchronous).